bin/node/rpc/src/lib.rs

```diff
@@ -29,73 +29,131 @@
 	pub fetcher: Arc<F>,
 }
 
-impl<F> LightDeps<F> {
-	/// Create empty `LightDeps` with given `F` type.
-	///
-	/// This is a convenience method to be used in the service builder,
-	/// to make sure the type of the `LightDeps<F>` is matching.
-	pub fn none(_: Option<Arc<F>>) -> Option<Self> {
-		None
-	}
+/// extra dependencies for babe.
+pub struct BabeDeps {
+	/// babe config.
+	pub babe_config: Config,
+	/// shared reference to EpochChanges
+	pub shared_epoch_changes: SharedEpochChanges<Block, Epoch>,
+	/// shared reference to the Keystore
+	pub keystore: KeyStorePtr,
 }
 
-/// Instantiate all RPC extensions.
-///
-/// If you provide `LightDeps`, the system is configured for light client.
-pub fn create<C, P, M, F>(
-	client: Arc<C>,
-	pool: Arc<P>,
-	light_deps: Option<LightDeps<F>>,
-) -> jsonrpc_core::IoHandler<M> where
+/// Full client dependecies
+pub struct FullDeps<C, P, SC> {
+	/// client instance
+	pub client: Arc<C>,
+	/// txpool
+	pub pool: Arc<P>,
+	/// SelectChain instance
+	pub select_chain: SC,
+	/// babe specific dependencies.
+	pub babe: BabeDeps,
+}
+
+
+/// Instantiate all Full RPC extensions.
+pub fn create_full<C, P, M, SC>(
+	deps: FullDeps<C, P, SC>,
+) -> Result<jsonrpc_core::IoHandler<M>, io::Error> where
 	C: ProvideRuntimeApi<Block>,
-	C: sc_client::blockchain::HeaderBackend<Block>,
+	C: HeaderBackend<Block> + HeaderMetadata<Block, Error=BlockChainError> + 'static,
 	C: Send + Sync + 'static,
 	C::Api: substrate_frame_rpc_system::AccountNonceApi<Block, AccountId, Index>,
 	C::Api: pallet_contracts_rpc::ContractsRuntimeApi<Block, AccountId, Balance, BlockNumber>,
 	C::Api: pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<Block, Balance, UncheckedExtrinsic>,
-	F: sc_client::light::fetcher::Fetcher<Block> + 'static,
+	C::Api: BabeApi<Block>,
+	<C::Api as sp_api::ApiErrorExt>::Error: fmt::Debug,
 	P: TransactionPool + 'static,
 	M: jsonrpc_core::Metadata + Default,
+	SC: SelectChain<Block> +'static,
 {
-	use substrate_frame_rpc_system::{FullSystem, LightSystem, SystemApi};
+	use substrate_frame_rpc_system::{FullSystem, SystemApi};
 	use pallet_contracts_rpc::{Contracts, ContractsApi};
 	use pallet_transaction_payment_rpc::{TransactionPayment, TransactionPaymentApi};
 
 	let mut io = jsonrpc_core::IoHandler::default();
+	let FullDeps {
+		client,
+		pool,
+		select_chain,
+		babe
+	} = deps;
+	let BabeDeps {
+		keystore,
+		babe_config,
+		shared_epoch_changes,
+	} = babe;
+
+
+	io.extend_with(
+		SystemApi::to_delegate(FullSystem::new(client.clone(), pool))
+	);
+	// Making synchronous calls in light client freezes the browser currently,
+	// more context: https://github.com/paritytech/substrate/pull/3480
+	// These RPCs should use an asynchronous caller instead.
+	io.extend_with(
+		ContractsApi::to_delegate(Contracts::new(client.clone()))
+	);
+	io.extend_with(
+		TransactionPaymentApi::to_delegate(TransactionPayment::new(client.clone()))
+	);
+	io.extend_with(
+		BabeRPC::to_delegate(
```

Comment:
Not a fan of the casing and naming here, all other apis have `Api` suffix, and the module name or the crate name indicate it's RPC. I think it makes sense here as well, to keep the convention.
```
babe_rpc::BabeApi::to_delegate(babe_rpc::Babe::new(client, ...))
```